pkg/binary/envoy/controlplane/istio.tmpl_test.go

```diff
@@ -16,51 +16,25 @@ package controlplane
 
 import (
 	"fmt"
-	"io"
 	"io/ioutil"
 	"net/http"
-	"os"
-	"path/filepath"
 	"testing"
 
-	"github.com/mholt/archiver"
 	"github.com/stretchr/testify/require"
 )
 
 // TestIstioBootStrapTemplateEqualsReleaseJson ensures istioBootStrapTemplate is exactly the same as what would have been
 // downloaded from the istio release for version IstioVersion.
 func TestIstioBootStrapTemplateEqualsReleaseJson(t *testing.T) {
-	// Retrieve Istio tarball os doesn't matter we only car about the bootstrap JSON
-	url := fmt.Sprintf("https://github.com/istio/istio/releases/download/%s/istio-%s-linux.tar.gz", IstioVersion, IstioVersion)
+	url := fmt.Sprintf("https://raw.githubusercontent.com/istio/istio/%s/tools/packaging/common/envoy_bootstrap_v2.json", IstioVersion)
```

Comment:
currently, this tests IstioVersion (which resolves 1.2.5 at the moment). We actually import this:

```
	istio.io/istio v0.0.0-20200304114959-c3c353285578
```

I'm tempted to read this from the go modules cache, to avoid I/O to github on every test run, but...

* the istio version in our constant file doesn't align with go.mod
* I don't know a clean way to get a relative path to resources in another go module, yet.

Even if we could read this, I suspect it would drift as we wouldn't always import the version of istio/istio we expect people to run, so this is just a thought cc @nacx

Reply:
cc also @basvanbeek in case you know how to sneakily get an arbitrary file from an imported go modules's tree (in go)

Reply:
not from within go binary but using in project root: go list -f '{{ .Dir }}' -m github.com/istio.io/istio would get you the path to the downloaded dependency

Reply:
nm I noticed this whole source code and test were a substitute for go:embed

Reply:
thanks for the tip, btw!